picard/ui/searchdialog.py

```diff
@@ -0,0 +1,394 @@
+                QtGui.QAbstractItemView.SingleSelection)
+        self.setSelectionBehavior(
+                QtGui.QAbstractItemView.SelectRows)
+        self.setEditTriggers(
+                QtGui.QAbstractItemView.NoEditTriggers)
+
+        self.horizontalHeader().setStretchLastSection(True)
+        self.horizontalHeader().setResizeMode(
+                QtGui.QHeaderView.Stretch)
+        self.horizontalHeader().setResizeMode(
+                QtGui.QHeaderView.Interactive)
+
+
+class SearchBox(QtGui.QWidget):
+
+    def __init__(self, parent):
+        self.parent = parent
+        QtGui.QWidget.__init__(self, parent)
+        self.search_action = QtGui.QAction(icontheme.lookup('system-search'),
+                _(u"Search"), self)
+        self.search_action.triggered.connect(self.search)
+        self.refresh_action = QtGui.QAction(icontheme.lookup('view-refresh'),
+                _(u"Refresh"), self)
+        self.refresh_action.triggered.connect(self.parent.retry)
+        self.setupUi()
+
+    def setupUi(self):
+        self.layout = QtGui.QVBoxLayout(self)
+        self.search_row_widget = QtGui.QWidget()
+        self.search_row_layout = QtGui.QHBoxLayout(self.search_row_widget)
+        self.search_row_layout.setContentsMargins(1, 1, 1, 1)
+        self.search_row_layout.setSpacing(1)
+        self.search_edit = ButtonLineEdit(self.search_row_widget)
+        self.search_row_layout.addWidget(self.search_edit)
+        self.search_button = QtGui.QToolButton(self.search_row_widget)
+        self.search_button.setAutoRaise(True)
+        self.search_button.setDefaultAction(self.search_action)
+        self.search_button.setIconSize(QtCore.QSize(22, 22))
+        self.search_row_layout.addWidget(self.search_button)
+        self.refresh_button = QtGui.QToolButton(self.search_row_widget)
+        self.refresh_button.setAutoRaise(True)
+        self.refresh_button.setDefaultAction(self.refresh_action)
+        self.refresh_button.setIconSize(QtCore.QSize(22, 22))
+        self.search_row_layout.addWidget(self.refresh_button)
+        self.search_row_widget.setLayout(self.search_row_layout)
+        self.layout.addWidget(self.search_row_widget)
+        self.adv_opt_row_widget = QtGui.QWidget()
+        self.adv_opt_row_layout = QtGui.QHBoxLayout(self.adv_opt_row_widget)
+        self.adv_opt_row_layout.setAlignment(QtCore.Qt.AlignLeft)
+        self.adv_opt_row_layout.setContentsMargins(1, 1, 1, 1)
+        self.adv_opt_row_layout.setSpacing(1)
+        self.use_adv_search_syntax = QtGui.QCheckBox(self.adv_opt_row_widget)
+        self.use_adv_search_syntax.setText(_("Use advance query syntax"))
+        self.adv_opt_row_layout.addWidget(self.use_adv_search_syntax)
+        self.adv_syntax_help = QtGui.QLabel(self.adv_opt_row_widget)
+        self.adv_syntax_help.setOpenExternalLinks(True)
+        self.adv_syntax_help.setText(_(
+                "(<a href='https://musicbrainz.org/doc/Indexed_Search_Syntax'>"
```

Comment:
The spacing issue I mentioned is fixed if I add `&#160;` before the opening parentheses.